pkg/controller/utils/utils.go

```diff
@@ -273,3 +276,63 @@ func GetAmazonCloudIntegration(ctx context.Context, client client.Client) (*oper
 
 	return instance, nil
 }
+
+// getAuthentication
+func GetAuthentication(ctx context.Context, client client.Client) (interface{}, error) {
+	var authenticationConfig interface{}
+
+	authenticationCR := &operatorv1.Authentication{}
```

Comment:
Does this default everything correctly?  (It gets returned like this in the not found case.)

Reply:
This CR isn't returned, right now the only things that will be returned will be either nil or render.OIDCAuthentication. This just allows use to do a type switch to see if we should add OIDC configuration instead of doing a bunch of checks on the authentication CR.

This also prevents the need to change the function definitions each time we add an auth method. Nothing has really changed here from what was in the logstorage controller, this was just moved the the core controller could use the same logic.